core/scale/encoder/primitives.hpp

```diff
@@ -108,26 +116,46 @@ namespace kagome::scale {
       const auto val = math::toLE(v);
       putByte(func, (uint8_t *)&val, size);
     } else {
-      encode(func, utils::to_tuple_refs(v));
+      kagome::scale::encode(func, utils::to_tuple_refs(v));
     }
   }
 
-  template <typename F,
-            typename T,
-            std::enable_if_t<std::is_enum_v<std::decay_t<T>>, bool> = true>
-  constexpr void encode(const F &func, const T &value) {
-    encode(func, static_cast<std::underlying_type_t<std::decay_t<T>>>(value));
+  constexpr void encode(const Invocable auto &func, const IsEnum auto &value) {
+    kagome::scale::encode(
+        func,
+        static_cast<std::underlying_type_t<std::decay_t<decltype(value)>>>(
+            value));
   }
 
-  template <typename F, typename T, typename... Args>
-  constexpr void encode(const F &func, const T &t, const Args &...args) {
-    encode(func, t);
-    encode(func, args...);
+  template <typename T, typename... Args>
+  constexpr void encode(const Invocable auto &func,
+                        const T &t,
+                        const Args &...args) {
+    kagome::scale::encode(func, t);
+    kagome::scale::encode(func, args...);
   }
 
   template <typename... Args>
   outcome::result<std::vector<uint8_t>> encode(const Args &...args) {
-    return ::scale::encode(args...);
+    auto ref = ::scale::encode(args...).value();
+    std::vector<uint8_t> res;
+    kagome::scale::encode(
+        [&](const uint8_t *const val, size_t count) {
+          if (count != 0ull) {
+            res.insert(res.end(), &val[0], &val[count]);
+          }
+        },
+        args...);
+    if (res != ref) {
+      std::string s;
+      auto append = [&](const auto &a) {
+        s += "\n";
+        s += typeid(a).name();
+      };
+      (append(args), ...);
+      std::cout << "[Inconsistent data]: " << s << std::endl;
```

Comment:
No output is expected to std here

Reply:
If you insist then let's put a clarification:
```suggestion
      std::cout << "[scale encode inconsistent data]: " << s << std::endl;
```

Reply:
Why not to put an assert instead of just printing then?